module/validation/receipt_validator.go

```diff
@@ -96,6 +96,14 @@ func (v *receiptValidator) verifyChunksFormat(result *flow.ExecutionResult) erro
 	if result.Chunks.Len() != requiredChunks {
 		return engine.NewInvalidInputErrorf("invalid number of chunks, expected %d got %d", requiredChunks, result.Chunks.Len())
 	}
+
+	// We have at least one chunk, check chunk state consistency
+	chunks := result.Chunks.Items()
+	for i := range chunks[:len(chunks)-1] {
```

Comment:
```suggestion
	for i := range len(chunks) - 1 {
```
[As of Go 1.22](https://github.com/golang/go/issues/61405), you can range over a number directly.

Reply:
Hm, I recall trying this (range over a number) in a previous PR but it failed the linter. At the time I thought it was because I was using Go 1.23 locally and the linter was running on Go 1.22, and the language feature was only supported in Go 1.23; but if it was added in 1.22 maybe it's just the linter that's behind?

Reply:
in this commit: https://github.com/onflow/flow-go/pull/6809/commits/72e87038f5b2ced753833e91d7449ad4d6c3439a#file-engine-execution-ingestion-uploader-uploader-go-L39 the linter complained about "cannot range over untyped int constant" - maybe it was the fact that it wasn't explicitly typed, rather than the syntax itself?

Reply:
Interesting. Did a bit of digging and it seems like:
- the linter is compatible with Go versions up to the version it was compiled with
- older releases are not re-compiled with new Go versions

The linter version we use `v1.54.2` was compiled with go1.21, hence the lack of compatibility. We will need to update the linter version.
```
% ./golangci-lint version
golangci-lint has version 1.54.2 built with go1.21.0 from 411e0bbb on 2023-08-21T12:04:32Z
```